view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js

```diff
@@ -266,49 +262,68 @@ define(
 
                             var validate = $(form).validation() && $(form).validation('isValid');
 
-                            // if oneclick or recurring is a card do validation on expiration date
+                            // if oneclick or recurring is a card check CVC validity
+                            var cid = true;
                             if (this.agreement_data.card) {
-                                // add extra validation because jquery validation will not work on non name attributes
-                                var expiration = Boolean($(form + ' #' + codeValue + '_expiration').valid());
-                                var expiration_yr = Boolean($(form + ' #' + codeValue + '_expiration_yr').valid());
-
-                                // only check if recurring type is set to oneclick
-                                var cid = true;
-                                if (this.hasVerification()) {
-                                    var cid = Boolean($(form + ' #' + codeValue + '_cc_cid').valid());
+                                // if encrypted cvc is empty the request is not valid
+                                if (this.hasVerification() && this.encryptedCreditCardVerificationNumber.length === 0) {
+                                    cid = false;
                                 }
-                            } else {
-                                var expiration = true;
-                                var expiration_yr = true;
-                                var cid = true;
                             }
 
-                            if (!validate || !expiration || !expiration_yr || !cid) {
+                            if (!validate || !cid) {
                                 return false;
                             }
 
                             return true;
                         },
-                        selectExpiry: function () {
-                            updatedExpiryDate = true;
-                            var self = this;
-                            self.expiry(true);
-                            return true;
+                        getCode: function () {
+                            return self.item.method;
+                        },
+                        getLocale: function () {
+                            return window.checkoutConfig.payment.adyenOneclick.locale;
+                        },
+                        getOriginKey: function () {
+                            return window.checkoutConfig.payment.adyenOneclick.originKey;
+                        },
+                        getLoadingContext: function () {
+                            return window.checkoutConfig.payment.adyenOneclick.checkoutUrl;
+                        },
+                        getGenerationTime: function () {
```

Comment:
generationTime shouldn't be needed for Secured Fields initialisation. Could you confirm that removing it is fine?

Reply:
yes it's not necessary anymore :)